contracts/contracts/validator-registry/middleware/MevCommitMiddleware.sol

```diff
@@ -0,0 +1,607 @@
+    }
+
+    function _setVaultFactory(IRegistry _vaultFactory) internal {
+        require(_vaultFactory != IRegistry(address(0)), ZeroAddressNotAllowed());
+        vaultFactory = _vaultFactory;
+        emit VaultFactorySet(address(_vaultFactory));
+    }
+
+    /// @dev Internal function to set the network address, which must have registered with the NETWORK_REGISTRY.
+    function _setNetwork(address _network) internal {
+        require(networkRegistry.isEntity(_network), NetworkNotEntity(_network));
+        network = _network;
+        emit NetworkSet(_network);
+    }
+
+    /// @dev Internal function to set the slash period in seconds.
+    function _setSlashPeriodSeconds(uint256 slashPeriodSeconds_) internal {
+        slashPeriodSeconds = slashPeriodSeconds_;
+        emit SlashPeriodSecondsSet(slashPeriodSeconds_);
+    }
+
+    /// @dev Internal function to set the slash oracle.
+    function _setSlashOracle(address slashOracle_) internal {
+        slashOracle = slashOracle_;
+        emit SlashOracleSet(slashOracle_);
+    }
+
+    /// @dev Authorizes contract upgrades, restricted to contract owner.
+    // solhint-disable-next-line no-empty-blocks
+    function _authorizeUpgrade(address) internal override onlyOwner {}
+
+    function _checkOperator(address operator) internal view {
+        require(operatorRegistry.isEntity(operator), OperatorNotEntity(operator));
+        require(operatorRecords[operator].exists, OperatorNotRegistered(operator));
+        require(!operatorRecords[operator].deregRequestOccurrence.exists, OperatorDeregRequestExists(operator));
+        require(!operatorRecords[operator].isBlacklisted, OperatorIsBlacklisted(operator));
+    }
+
+    function _checkCallingOperator(address operator) internal view {
+        require(msg.sender == operator, OnlyOperator(operator));
+        _checkOperator(operator);
+    }
+
+    function _checkVault(address vault) internal view {
+        require(vaultFactory.isEntity(vault), VaultNotEntity(vault));
+        require(vaultRecords[vault].exists, VaultNotRegistered(vault));
+        require(!vaultRecords[vault].deregRequestOccurrence.exists, VaultDeregRequestExists(vault));
+    }
+
+    /// @dev Returns the one-indexed position of the blsPubkey in the set.
+    /// @return 0 if the blsPubkey is not in the set.
+    function _getPositionInValset(bytes calldata blsPubkey,
+        address vault, address operator) internal view returns (uint256) {
+        return _vaultAndOperatorToValset[vault][operator].position(blsPubkey);
+    }
+
+    function _isValidatorReadyToDeregister(bytes calldata blsPubkey) internal view returns (bool) {
+        return validatorRecords[blsPubkey].deregRequestOccurrence.exists && 
+            block.timestamp > slashPeriodSeconds + validatorRecords[blsPubkey].deregRequestOccurrence.timestamp;
+    }
+
+    function _isOperatorReadyToDeregister(address operator) internal view returns (bool) {
```

Comment:
It will be a little bit more effective to save `operatorRecords[operator]` to the local variable (`operatorRecord`)